File: src/environment/check_key.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   check_key.c                                        :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jschwabe <jschwabe@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2024/03/29 18:32:12 by jschwabe          #+#    #+#             */
/*   Updated: 2024/03/29 18:32:13 by jschwabe         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include "libft.h"
#include <stdbool.h>

bool	check_valid_key(const char *s)
{
	int	i;

	i = 0;
	if (*s == '=' || !(ft_isalpha(*s) || *s == '_'))
		return (false);
	while (s[i] && s[i] != '=')
	{
		if (ft_isalnum(s[i]) || s[i] == '_')
			i++;
		else
			return (false);
	}
	if (s[i] == '=' && s[i + 1] == '=')
		return (false);
	return (true);
}

File: src/environment/check_key.rs
pub fn check_valid_key(s: &[u8]) -> bool {
	let mut i = 0;
	if s[0] == b'=' || !(matches!(s[0] as char,'A'..='Z' | 'a'..='z' | '_')) {
		return false;
	}
	while s[i] != 0 && s[i] != b'=' {
		if matches!(s[i] as char,'A'..='Z' | 'a'..='z' | '0'..='9' | '_') {
			i += 1;
		} else {
			return false;
		}
	}
	!(s[i] == b'=' && s[i + 1] == b'=')
}
